pymeasure/instruments/temptronic/temptronic_eco560.py

```diff
@@ -0,0 +1,86 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""
+Implementation of an interface class for ThermoStream® Systems devices.
+Reference Document for implementation:
+ECO-560/660 ThermoStream® Systems
+Operators Manual
+Revision B
+September, 2018
+"""
+
+from pymeasure.instruments.temptronic.temptronic_base import ATSBase
+from pymeasure.instruments.instrument import Instrument
+from pymeasure.instruments.validators import truncated_range
+
+#
+# NO_DUT_SENSOR_SELECTED = 16384      # bit 14 – no DUT sensor selected
```

Comment:
To account for these differences, you could put these values in an `Eco560ErrorCode` IntFlag, turn `ATSBase.error_code` into a dynamic property (see below), and redefine its `get_process` in ECO560 to use the correct error code class.

Reply:
Ive started running this implementation but am running into issues when assigning the lamba statement to the get process. Its suggesting converting it into a fucntion implementation. Any ideas/examples of how to get around this? Im also running into issues with the hp legacy instruments tests after the most recent commits which I have pulled in, these are not relevant to this pull request but are preventing successful CI completion

Reply:
> Its suggesting converting it into a fucntion implementation.

The warning means that style-wise, it's weird to take a lambda (a "temporary"-ish function) and assign it a permanent name.
In this instance it's fine, you should be able to quiet the warning by adding a `#noqa: E731` comment.

We're aware of the HP Legacy instruments tests problem, will be fixed.

Reply:
Thanks! Also done

Reply:
I just now merged a fix for the CI. If it still comes up red for the same reason as before, you'll need to force push or otherwise update your branch to trigger a run with a "fresh" base branch.